src/java.base/share/classes/java/lang/Class.java

```diff
@@ -3066,18 +3066,12 @@ private static void checkPackageAccessForPermittedSubclasses(SecurityManager sm,
     private String resolveName(String name) {
         if (!name.startsWith("/")) {
```

Comment:
I expect this would be more more readable if you invert it, i.e. "if (name.startsWith("/") { return name.substring(1); } else { ... }.

Note that the baseName == null check was needed when it was originally created because getPackageName could return null in the initial version.

Reply:
@AlanBateman As Peter commented below there's likely to be no improvement when the code is called from `j.l.Class` itself, and indeed there's no any. 

However, there's still unnecessary null check here and in other places, so is it reasonable to reword the ticker and this PR to rid that check, or create another ticket, or it's not worth the effort? What do you thinl?